jina/logging/formatter.py

```diff
@@ -29,9 +29,10 @@ def format(self, record):
         :param record: A LogRecord object
         :returns: Formatted LogRecord with level-colour MAPPING to add corresponding colour.
         """
-        cr = copy(record)
+        cr = deepcopy(record)
```

Comment:
hey @chunyuema,

Why do we need deepcopy in this case?

Reply:
Hi, thank you for the question. I suggested using deepcopy mostly because the record is a LogRecord object. Making a copy of the object should avoid most of the accidental overwrite of the original copy. However, I was a little worried on effect of codes from line 88-89 on the original object. I think even though it seems to be sufficient to just use copy for now, it might still be a better practice to make a completely independent copy of the LogRecord object into cr so that the integrity of the original copy would be preserved as much as possible. This might become more relevant if more operations are added with the cr variable.

Reply:
Hi @chunyuema, will deepcopy harm the performance?

Reply:
Hi, this is a good question. I think deepcopy will definitely be slower than just copy. I need to do more study on the code base to see how much using deepcopy impacts the performance.

Reply:
Sure. Since the `formatter` is used in `logger`, which is used extensively. I'm not sure if this will harm the performance, haven't got time to look into it. To my understanding, we should use `deepcopy` only when necessary.

And it's good that you can raise this suggestion.